File: HeartForCharity/UI/heartforcharity_desktop/lib/model/search_objects/volunteer_job_search_object.dart
import 'package:heartforcharity_desktop/model/search_objects/base_search_object.dart';

class VolunteerJobSearchObject extends BaseSearchObject {
  final int? categoryId;
  final String? status;
  final int? organisationProfileId;
  final bool? isRemote;
  final int? cityId;

  VolunteerJobSearchObject({
    super.fts,
    super.page,
    super.pageSize,
    super.includeTotalCount,
    super.retrieveAll,
    this.categoryId,
    this.status,
    this.organisationProfileId,
    this.isRemote,
    this.cityId,
  });

  @override
  Map<String, dynamic> toMap() => {
        ...super.toMap(),
        if (categoryId != null) 'categoryId': categoryId,
        if (status != null) 'status': status,
        if (organisationProfileId != null)
          'organisationProfileId': organisationProfileId,
        if (isRemote != null) 'isRemote': isRemote,
        if (cityId != null) 'cityId': cityId,
      };
}

File: HeartForCharity/HeartForCharity.Model/SearchObjects/VolunteerJobSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HeartForCharity.Model.SearchObjects
{
    public class VolunteerJobSearchObject : BaseSearchObject
    {
        public int? CategoryId { get; set; }
        public string? Status { get; set; }
        public int? OrganisationProfileId { get; set; }
        public bool? IsRemote { get; set; }
        public int? CityId { get; set; }
        public DateTime? StartDateFrom { get; set; }
        public DateTime? StartDateTo { get; set; }
    }
}
